include/blast/math/algorithm/Trmm.hpp

```diff
@@ -67,159 +71,256 @@ namespace blast
             }
         }
     }
-    /// @brief C = alpha * A * B + C; A upper-triangular
+
+
+    /// @brief C = alpha * A * B; A upper- or lower-triangular. Matrix pointer arguments.
     ///
-    template <typename ST, typename MT1, typename MT2, typename MT3>
-    requires Matrix<MT1, ST> && Matrix<MT2, ST> && Matrix<MT3, ST>
-        && (StorageOrder_v<MT1> == columnMajor) && (StorageOrder_v<MT3> == columnMajor)
-    inline void trmmLeftUpper(
-        ST alpha,
-        MT1 const& A, MT2 const& B,
-        MT3& C)
+    /// See https://netlib.org/lapack/explore-html-3.6.1/d1/d54/group__double__blas__level3_gaf07edfbb2d2077687522652c9e283e1e.html
+    ///
+    /// @tparam MPA matrix pointer type for matrix A
+    /// @tparam MPB matrix pointer type for matrix B
+    /// @tparam MPC matrix pointer type for matrix C
+    ///
+    /// @param M the number of rows of B
+    /// @param N the number of columns of B
+    /// @param alpha the scalar alpha
+    /// @param A pointer to top left element of matrix A
+    /// @param uplo specifies whether the matrix A is an upper or lower triangular
+    /// @param diag specifies whether or not A is unit triangular
+    /// @param B pointer to top left element of matrix B
+    /// @param C pointer to top left element of matrix C
+    ///
+    template <typename ST, typename MPA, typename MPB, typename MPC>
+    requires MatrixPointer<MPA, ST> && MatrixPointer<MPB, ST> && MatrixPointer<MPC, ST>
+        && (StorageOrder_v<MPA> == columnMajor) && (StorageOrder_v<MPC> == columnMajor)
+    inline void trmm(size_t M, size_t N, ST alpha, MPA A, UpLo uplo, bool diag, MPB B, MPC C)
     {
         using ET = ST;
         size_t constexpr TILE_SIZE = TileSize_v<ET>;
 
-        size_t const M = rows(B);
-        size_t const N = columns(B);
-
-        if (rows(A) != M || columns(A) != M)
-            throw std::invalid_argument {"Matrix sizes do not match"};
+        if (diag)
```

Comment:
This should probably `diagonalUnit` or so. Now it sounds like this means 'diagonal matrix'.

Reply:
Why didn't you go for a type like with `UpLo`?

Reply:
> Why didn't you go for a type like with `UpLo`?

Probably because "diagonal is unit" is a boolean property, while assigning `true` or `false` to "upper
 and "lower" is arbitrary. What do you think?